lib/protobuf.ex

```diff
@@ -62,17 +62,24 @@ defmodule Protobuf do
           Foo.new_and_verify!(a: 123)
 
       would raise an exception.
+
+      It does no verification in production.
       """
       def new_and_verify!(attrs) do
         struct = Protobuf.Builder.new!(__MODULE__, attrs)
 
-        case Protobuf.Verifier.verify(struct) do
-          {:error, messages} ->
-            raise Protobuf.VerificationError, message: Enum.join(messages, "\n\t")
-
-          :ok ->
-            struct
+        quote do
+          if Mix.env != :prod do
```

Comment:
instead of matching on `Mix.env != :prod` can we have this be a config option? Similarly to how you can enable extensions maybe something like the below setting and then we can check if it's enabled or not here?
```
config :protobuf,
  verify: :enabled
```

Reply:
Config options are definitely a step up from checking `Mix.env != :prod`

We talked a little bit about whether having this be magical is a good idea. i.e. we could even put this code block inside `new!` so that everyone's test file protobuf instantiations would automatically start getting verified

Overall I think we want to keep it simple - we'll merge new_and_verify! as a straightfoward function that doesn't change behavior based on environment or config. And we'll try and use the linter to keep new_and_verify! out of non-test files, but encourage it in test files